hs/src/Reach/Connector/ALGO.hs

```diff
@@ -3199,9 +3225,20 @@ cmeth sigi = \case
 
 bindFromArgs :: [DLVarLet] -> App a -> App a
 bindFromArgs vs m = do
-  -- XXX deal with >15 args
-  let go (v, i) = sallocVarLet v False (code "txna" ["ApplicationArgs", texty i] >> cfrombs (varLetType v))
-  foldl' (flip go) m (zip vs [(1::Integer) ..])
+  let goSingle (v, i) = sallocVarLet v False (code "txna" ["ApplicationArgs", texty i] >> cfrombs (varLetType v))
+  let goSingles singles k =
+        foldl' (flip goSingle) k (zip singles [(1 :: Integer) ..])
+  -- When there are more than 15 args, the 15th arg is a tuple of args 15+
+  case 15 < (length vs) of
+    False -> do
+      goSingles vs m
+    True -> do
+      let (vs14, vsMore) = splitAt 14 vs
```

Comment:
This pattern is so common. I think I want a function like:
```
splitArgs :: [a] -> ([a], Maybe [a])
splitArgs l =
 case 15 < (length l) of
   False -> (l, Nothing)
   True -> (before, Just after) where (before, after) = splitAt 14 l
```
and then adapt the code to only branch on the Maybe rather than the list. I think it will de-duplicate some code